capstone/src/main/webapp/script.js

```diff
@@ -289,15 +290,16 @@ function mapSentiment(colorMap) {
     {fillColor: precinctColor, fillOpacity: 0.9});
   }
 }
-
+//* * get precinct name as parameter and returns precinct ID*/
 function searchPrecinctsByDistrict(desiredDistrict) {
-  mapAndSelection.map.forEachFeature( function(feature) {
-    if (feature.getProperty('district') == desiredDistrict) {
-      return feature.getProperty('id');
-    }
-  });
+    let precinctLayer = mapAndSelection.map;
+    precinctLayer.forEach((precinct) => {
+        if (precinct.getProperty('district') == desiredDistrict) {
```

Comment:
Its really interesting that the return here ends up returning from the external function! I didn't know that would happen, its an interesting behavior. You might also want to consider using the find method, since it might make the code slightly more self documenting.